### PR TITLE
plantuml-server: init at 1.2020.14

### PR DESCRIPTION
**Pull request by truh**
###### Motivation for this change
Add PlantUML server package and service module. https://plantuml.com/server

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/plantuml-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "plantuml-server";
+  version = "1.2020.15";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/project/plantuml/${version}/plantuml.${version}.war";
```
Comment (sikmir) [inline pkgs/tools/misc/plantuml-server/default.nix:8]: * It would be better to build .war from source
* There is https://github.com/plantuml/plantuml-server

File: pkgs/tools/misc/plantuml-server/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  };
+
+  buildCommand = ''
+    mkdir -p "$out/webapps"
+    cp "$src" "$out/webapps/plantuml.war"
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A web application to generate UML diagrams on-the-fly.";
+    homepage = https://plantuml.com/;
```
Comment (sikmir) [inline pkgs/tools/misc/plantuml-server/default.nix:19]: Unquoted URL syntax is deprecated by [RFC 45](https://github.com/NixOS/rfcs/pull/45):
```suggestion
    homepage = "https://plantuml.com/";
```

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -8170,6 +8170,12 @@
     githubId = 483735;
     name = "Dmitry Geurkov";
   };
+  truh = {
```
Comment (sikmir) [inline maintainers/maintainer-list.nix:8173]: IIRC, adding yourself as a maintainer is preferred to be a separate commit.
Something like, "maintainers: add truh". Then apply the plantuml-server commit.

File: pkgs/tools/misc/plantuml-server/default.nix (hunk 1 above)
Comment (truh) [reply pkgs/tools/misc/plantuml-server/default.nix]: done

File: pkgs/tools/misc/plantuml-server/default.nix (hunk 2 above)
Comment (truh) [reply pkgs/tools/misc/plantuml-server/default.nix]: done

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (truh) [reply maintainers/maintainer-list.nix:8173]: done

**Review by sikmir — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 92582` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>plantuml-server</li>
  </ul>
</details>

**Review by doronbehar — APPROVED** [review]
Diff LGTM. Tested with `services.plantuml-server.enable = true;` and it seems functional. Thanks for the contribution :).
